### PR TITLE
Fix dependency on filesystem on Linux

### PR DESCRIPTION
**Pull request by atalman**
Similar to: https://github.com/pytorch/pytorch/pull/134494
We are seeing come back of https://github.com/pytorch/pytorch/issues/133437 due to use of filesystem on Linux

### CODE REVIEW

File: torch/csrc/inductor/aoti_torch/shim_common.cpp [hunk 1]
```diff
@@ -57,16 +57,24 @@ namespace fs = std::experimental::filesystem;
...
+std::string get_current_path() {
+#ifdef _WIN32
+  return fs::current_path().string();
 #else
-  throw std::runtime_error("Not implemented");
+  char currentPath[PATH_MAX];
+  if (getcwd(currentPath, sizeof(currentPath)) != nullptr) {
+    return std::string(currentPath);
+  } else {
+    throw std::runtime_error("Failed to get current path");
```
Comment (kit1980) [inline torch/csrc/inductor/aoti_torch/shim_common.cpp:76]: When this is possible?

**Review by kit1980 — APPROVED** [review]

File: torch/csrc/inductor/aoti_torch/shim_common.cpp (hunk 1 above)
Comment (malfet) [reply torch/csrc/inductor/aoti_torch/shim_common.cpp:76]: When your current working directory is deleted for example

File: torch/csrc/inductor/aoti_torch/shim_common.cpp [hunk 2]
```diff
@@ -57,16 +57,24 @@ namespace fs = std::experimental::filesystem;
...
 #endif
 
 // HACK for failed builds in ARVR, where it cannot find these symbols within
 // std::experimental::filesystem
 namespace {
-fs::path get_current_path() {
-#if __has_include("filesystem")
-  return fs::current_path();
+std::string get_current_path() {
+#ifdef _WIN32
```
Comment (malfet) [inline torch/csrc/inductor/aoti_torch/shim_common.cpp:69]: I would change it to `#if __has_include("filesystem") && !defined(__LINUX__)` as we want to use those on both Windows and Mac, where we don't have this problem

File: torch/csrc/inductor/aoti_torch/shim_common.cpp [hunk 3]
```diff
@@ -80,10 +88,13 @@ bool file_exists(std::string& path) {
 }
 
 bool create_directories(const std::string& path) {
-#if __has_include("filesystem")
+#ifdef _WIN32
```
Comment (malfet) [inline torch/csrc/inductor/aoti_torch/shim_common.cpp:91]: Same as above, don't restrict it to windows and check that CXX11 ABI is present

**Review by malfet — APPROVED** [review]

File: torch/csrc/inductor/aoti_torch/shim_common.cpp (hunk 1 above)
Comment (malfet) [reply torch/csrc/inductor/aoti_torch/shim_common.cpp:76]: See https://godbolt.org/z/4on5Y6nsG

**Comment by kit1980** [comment]
Manually cherry-picking into 2.5 - https://github.com/pytorch/pytorch/pull/137242

**Pull request closed without merging**